synapse/datamodel.py

```diff
@@ -580,6 +580,8 @@ def __init__(self):
                 ('depth', ('int', {}), {'ro': True, 'doc': 'The depth indexed from 0.', }),
                 ('parent', ('$self', {}), {'ro': True, 'doc': 'The taxonomy parent.', }),
             ),
+            'doc': 'Deprecated. Please use meta:taxonomy.',
```

Comment:
Is there a migration path forward for someone with an extended model type that uses the old `taxonomy` interface which does not require them to store data off to the side somewhere in order to update their custom type to `meta:taxonomy` ?

Reply:
Yeah I think we could make a migration for that pretty easily, would we want to just migrate by default or have it be something that a user would need to trigger?

Reply:
There are pros and cons either way.

If we do a migration on behalf of users, the data model for their cortex would be consistent and users would avoid any deprecation notices. The downside is that any user which manages records of their extended properties outside of the Cortex would have a deviation. I am aware of users that manage extended properties like that; but I am not aware of any users which are using the taxonomy interface for extended forms.

If we offered it as a hot-fix we could document that and advise users to utilize it as needed. That would allow someone to opt-into the update and avoid the problem of moving data back and forth.